### PR TITLE
Remove `Copy` from `ResolverOpts`

### PR DESCRIPTION
**Pull request by daxpedda**
As discussed in https://github.com/bluejekyll/trust-dns/pull/2001#issuecomment-1725255970.

### CODE REVIEW

**Comment by daxpedda** [comment]
Was just looking at the CI issue here. This was fixed upstream in `time` in https://github.com/time-rs/time/pull/622. The only way I see how we can fix this right now is to either use an older nightly version or use [`--no-cfg-coverage-nightly`](https://github.com/taiki-e/cargo-llvm-cov/tree/5d08e207e49516a6ec84ee179a6fdf1965e50a5b?tab=readme-ov-file#exclude-function-from-coverage). Happy to make a separate PR!

**Comment by bluejekyll** [comment]
I was hoping there would be an update to time to resolve this, but looks like not yet: https://github.com/time-rs/time/issues/618#issuecomment-1726220929

File: crates/resolver/src/config.rs [hunk 1]
```diff
@@ -848,13 +848,13 @@ impl Default for ServerOrderingStrategy {
...
 /// Configuration for the Resolver
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[derive(Debug, Clone, Eq, PartialEq)]
 #[cfg_attr(
     feature = "serde-config",
     derive(Serialize, Deserialize),
     serde(default)
 )]
-#[allow(dead_code)] // TODO: remove after all params are supported
+#[allow(dead_code, missing_copy_implementations)] // TODO: remove after all params are supported
```
Comment (bluejekyll) [inline crates/resolver/src/config.rs:857]: The TODO comment should probably be removed at this point...

**Review by bluejekyll — CHANGES_REQUESTED** [review]
It's a pretty minor change, but it probably worth cleaning that up.

File: crates/resolver/src/config.rs (hunk 1 above)
Comment (daxpedda) [reply crates/resolver/src/config.rs]: Done.

**Comment by daxpedda** [comment]
> I was hoping there would be an update to time to resolve this, but looks like not yet: [time-rs/time#618 (comment)](https://github.com/time-rs/time/issues/618#issuecomment-1726220929)

I'm happy to make a PR if I was told how to proceed:
1. Pin a working nightly version.
2. Use [`--no-cfg-coverage-nightly`](https://github.com/taiki-e/cargo-llvm-cov/tree/5d08e207e49516a6ec84ee179a6fdf1965e50a5b?tab=readme-ov-file#exclude-function-from-coverage).

**Comment by bluejekyll** [comment]
We can merge with coverage failing, it's just an annoyance, but I'm not concerned at this point.

**Review by bluejekyll — APPROVED** [review]
Looks great, thanks for the PR! (btw, this is technically a breaking change, so I think will go out in 0.24 which is currently not planned)

**Pull request merged**